coveralls/api.py

```diff
@@ -144,7 +146,36 @@ def load_config_from_semaphore():
     def load_config_from_unknown():
         return 'coveralls-python', None, None, None
 
+    def load_config_from_generic_ci_environment(self):
+        # Inspired by the official client:
+        # coveralls-ruby in lib/coveralls/configuration.rb
+        # (set_standard_service_params_for_generic_ci)
+
+        # The meaning of each env var is clarified in:
+        # https://github.com/lemurheavy/coveralls-public/issues/1558
+
+        config = {
+            'service_name': os.environ.get('CI_NAME'),
+            'service_number': os.environ.get('CI_BUILD_NUMBER'),
+            'service_build_url': os.environ.get('CI_BUILD_URL'),
+            'service_job_id': os.environ.get('CI_JOB_ID'),
+            'service_branch': os.environ.get('CI_BRANCH'),
+        }
+
+        pr_match = NUMBER_REGEX.findall(os.environ.get('CI_PULL_REQUEST', ''))
+        if pr_match:
+            config['service_pull_request'] = pr_match[-1]
```

Comment:
I think this is a bit different than [the ruby implementation](https://github.com/lemurheavy/coveralls-ruby/blob/6453624810be9862842874f0c812c7af45c5cadb/lib/coveralls/configuration.rb#L64).

I'm not too familiar with ruby, but from reading through the [docs on regex-based indexing](https://ruby-doc.org/core-3.0.2/String.html#method-i-5B-5D) is seems as if they're grabbing the _first_ numeric sequence, whereas you're grabbing the last numeric sequence here.

Reply:
I believe that it should not make a difference, since the regex ends in `$`... With this regex there should be only one match (which will be at the same time the first and the last).

_(I am assuming some CIs could export the entire URL, or part of it, as the `CI_PULL_REQUEST` env var, instead of just a number, and that the name of the repository can contain numbers (e.g. "john1988/myrepo"))_

![image](https://user-images.githubusercontent.com/320755/126217952-5102fd26-bcb0-4ab3-82ce-a68051e8736f.png)

If the `$` was not in the regex, then I believe the `-1` would be a slightly more robust option.

![image](https://user-images.githubusercontent.com/320755/126218032-cfb96e57-9662-4fbc-83d7-13c2008f4722.png)

Now that I wrote this comment, I wonder if I should remove the `$` from the regex... It is not uncommon to have URL ending with a trailing `/`...

Reply:
Hmm, fair enough. Yeah, it occured to me as I read your comment that I have no idea what set of values are expected to be contained in that env var. The description in the linked PR isn't all that specific...

This implementation makes sense to me, then -- but would you mind updating the configuration docs you added to clarify this env var? eg. to include something along the lines of "expected formats for this variable include..." or such?

Reply:
No problems @TheKevJames, I updated the description. Hopefully it is a bit better now? 🤞 

I also rebased the latest changes from the master.